### PR TITLE
GEODE-4035: Refactor Converter classes

### PR DESCRIPTION
**Pull request by jdeppe-pivotal**
- This specifically avoids use of the Spring String Converter

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/test/java/org/apache/geode/management/internal/cli/converters/ClusterMemberIdNameConverterJUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+  @Test
+  public void complete() throws Exception {
+    CommandCandidate candidate = parser.complete("describe member --name=");
+    assertThat(candidate.size()).isEqualTo(allMemberNames.length);
+    assertThat(candidate.getFirstCandidate()).isEqualTo("describe member --name=server1");
+
+    candidate = parser.complete("describe member --name=");
+    assertThat(candidate.size()).isEqualTo(2);
+    assertThat(candidate.getFirstCandidate()).isEqualTo("describe member --name=server1");
```
Comment (pdxrunner) [inline geode-core/src/test/java/org/apache/geode/management/internal/cli/converters/ClusterMemberIdNameConverterJUnitTest.java]: What coverage does this case add? Only difference between the two cases is the hard-coded expected value.

Comment (jinmeiliao) [reply geode-core/src/test/java/org/apache/geode/management/internal/cli/converters/ClusterMemberIdNameConverterJUnitTest.java]: a more extended test would be renaming this test to be BaseStringConverterTest, and spy all the converters you modified in this change set and try to do a parser.complete of all the affected parameters.

Comment (jdeppe-pivotal) [reply geode-core/src/test/java/org/apache/geode/management/internal/cli/converters/ClusterMemberIdNameConverterJUnitTest.java]: Right - I had added this test initially specifically for `describe member` but then started down the path of refactoring all the other pieces.

**Comment by jdeppe-pivotal** [comment]
FYI, there's still a bunch of work left to do here to tidy things up. Wherever the Member/Group converters are referenced in commands **and** associated with an array parameter, they will not work as the default `ArrayConverter` will get invoked.

**Pull request merged**